itou/users/admin.py

```diff
@@ -381,6 +377,26 @@ def get_readonly_fields(self, request, obj=None):
                 rof += ("first_name", "last_name", "email")
         return rof
 
+    def get_search_fields(self, request):
+        search_fields = []
+        search_term = request.GET.get("q", "").strip()
+        if len(search_term) == 30:
+            try:
+                int(search_term, base=16)
+            except ValueError:
+                pass
+            else:
+                search_fields.append("asp_uid__exact")
+        if search_term.isdecimal():
+            search_fields.append("pk__exact")
+            search_fields.append("nir__exact")
```

Comment:
_nit_ : Pour éviter de se poser la question "Pourquoi en deux fois ? Code dupliqué ? Problème de conflit ?" :grin: 
```suggestion
            search_fields.extend(["pk__exact", "nir__exact"])
```

Reply:
Je laisse, ça fait des opérations en plus que ce dont on a vraiment besoin.

Reply:
Parce que j’aime vraiment `timeit` :stuck_out_tongue: :
```
$ python -m timeit -s 'x=[1,2]' 'x.extend([3,4])'
5000000 loops, best of 5: 49.1 nsec per loop
$ python -m timeit -s 'x=[1,2]' 'x.append(3);x.append(4)'
10000000 loops, best of 5: 21.3 nsec per loop
```

Reply:
> Premature optimization is the root of all evil 

;)